crates/node-maintainer/src/maintainer.rs

```diff
@@ -140,14 +167,24 @@ impl NodeMaintainer {
     }
 
     async fn run_resolver(&mut self) -> Result<(), NodeMaintainerError> {
-        let mut package_groups = Vec::new();
+        let (package_sink, package_stream) = futures::channel::mpsc::unbounded();
         let mut q = VecDeque::new();
         q.push_back(self.graph.root);
+        let mut in_flight = 0;
+
+        let mut package_stream = package_stream
+            .map(|dep: NodeDependency| {
+                self.nassun
+                    .resolve(format!("{}@{}", dep.name, dep.spec))
+                    .map_ok(move |p| (p, dep))
+            })
+            .buffer_unordered(self.parallelism)
+            .ready_chunks(self.parallelism);
```

Comment:
If `ready_chunks()` is absent - the `q` will never have length greater than `1` and the sort does nothing.